packages/js/plugin/src/PluginPackage.ts

```diff
@@ -50,8 +50,8 @@ export class PluginPackage<
     }
   }
 
-  async getManifest(): Promise<Result<WrapManifest, Error>> {
-    return ResultOk(this._manifest);
+  async getManifest(): Promise<Result<Readonly<WrapManifest>, Error>> {
+    return ResultOk(JSON.parse(JSON.stringify(this._manifest)));
```

Comment:
I don't know if there is a node equivalent, but the JS Web API also has `structuredClone`: https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

Reply:
I agree with you. Found this snippet which seems to be extremely efficient (see `c7x43t`'s `deepcopy`): https://jsben.ch/gtKdJ

Adapted it slightly for Typescript and used that instead of JSON.stringify + parse

Reply:
while i agree that we should think of performance, I am not sure at which point we should stop thinking about it. My main thought is: Is the deep clone's speed really impactful when building or invoking a wrapper?

my concern is that here we are adding custom code which we're committing to maintaining; I think that we might appreciate in the future going with the approach of `structuredClone` or `JSON.parse(JSON.stringify(..))` because it is **way** much easier to debug. In cases like this, we will need to decide whether we want performance or maintainability, and in this scenario, I would go with the latter.

These are my two cents hehe no strong opinion on this. (My only strong opinion is that if we want to add that `deepCopy` function, we should add tests to it :P)

Reply:
That's a good point. There is a tradeoff here. How often is `getManifest` called?

Reply:
I thought about it more and you're right. I think it makes sense to use the simpler code. Sorry for complicating this!